cli/command/environment.go

```diff
@@ -236,18 +252,41 @@ func (e *EnvironmentCommand) link(c *cli.Context) error {
 		return fmt.Errorf("Cannot link an environment to itself")
 	}
 
-	req := models.CreateEnvironmentLinkRequest{
-		SourceEnvironmentID: sourceEnvironmentID,
-		DestEnvironmentID:   destEnvironmentID,
+	updateEnvReq, err := e.generateAddLinkRequest(sourceEnvironmentID, destEnvironmentID)
```

Comment:
I think we can make this quite a bit more readable and DRY with some helper functions. Maybe something like:
```
func (e *EnvironmentCommand) addLink(c *cli.Context) error {
	// ignore duplicates for bi-directional idempotence
	generateRequest := func(links []string, destEnvironmentID string) models.UpdateEnvironmentRequest {
		links = append(links, destEnvironmentID)

		return models.UpdateEnvironmentRequest{
			Links: &links,
		}
	}

	return e.updateEnvironmentLinksHelper(c, generateRequest)
}

func (e *EnvironmentCommand) removeLink(c *cli.Context) error {
	// ignore missing for bi-directional idempotence
	generateRequest := func(links []string, destEnvironmentID string) models.UpdateEnvironmentRequest {
		for i, link := range links {
			if link == destEnvironmentID {
				links = append(links[:i], links[i+1:]...)
				i--
			}
		}

		return models.UpdateEnvironmentRequest{
			Links: &links,
		}
	}

	return e.updateEnvironmentLinksHelper(c, generateRequest)
}

func (e *EnvironmentCommand) updateEnvironmentLinksHelper(
	c *cli.Context,
	generateRequest func(links []string, destEnvironmentID string) models.UpdateEnvironmentRequest,
) error {
	args, err := parseArgs(c, "SOURCE", "DEST")
	if err != nil {
		return err
	}

	sourceEnvironmentID, err := resolveSingle(args["SOURCE"])
	if err != nil {
		return err
	}

	destEnvironmentID, err := resolveSingle(args["DEST"])
	if err != nil {
		return err
	}

	updateFN := func(sourceEnvironmentID, destEnvironmentID string) error {
		sourceEnvironment, err := c.Client.ReadEnvironment(sourceEnvironmentID)
		if err != nil {
			return err
		}

		req := generateRequest(sourceEnvironment.Links, destEnvironmentID)
		jobID, err := e.Client.UpdateEnvironment(sourceEnvironmentID, req)
		if err != nil {
			return
		}

		return waitOnJobHelper(c, jobID, "updating environment", func(string) error {
			e.printer.Printf("done")
			return nil
		})
	}

	if err := updateFN(sourceEnvironmentID, destEnvironmentID); err != nil {
		return err
	}

	if c.Bool("uni-directional") {
		return nil
	}

	return updateFN(destEnvironmentID, sourceEnvironmentID)
}
```

Reply:
I like this 👍